lib/tapioca/dsl/compiler.rb

```diff
@@ -9,45 +9,34 @@
 
 module Tapioca
   module Dsl
-    DSL_COMPILERS_DIR = T.let(File.expand_path("../compilers", __FILE__).to_s, String)
-
     class Compiler
       extend T::Sig
       extend T::Helpers
+      extend T::Generic
 
       include Reflection
+      extend Reflection
```

Comment:
Is it intentional that we're including and extending the same module? Is there any way we could consolidate these?

Reply:
This is intentional, since we want `gather_constants` methods to have access to `Reflection` methods too, but that method is on the singleton class now.

Reply:
Not sure if there is a better way to do this, though.

Reply:
Yeah, I guess it'll be hard to escape this. Just checking.